geode-core/src/test/java/org/apache/geode/distributed/internal/InternalConfigurationPersistenceServiceTest.java

```diff
@@ -265,6 +270,26 @@ public void removeInvalidXmlConfiguration(String xml, int expectedInitialElement
         .isEqualTo(expectFinalElements);
   }
 
+  @Test
+  public void dontUnlockSharedConfigurationIfNoLockedPreviously() {
+    InternalConfigurationPersistenceService service3;
+
+    MetricsService.Builder metricsSessionBuilder = mock(MetricsService.Builder.class);
```

Comment:
looks like you don't need all these code to build ds, and you also don't need that extra "visisble for testing" constructor you added. You can just do: 


    JAXBService jaxbService = mock(JAXBService.class);
    InternalConfigurationPersistenceService cps = spy(new InternalConfigurationPersistenceService(jaxbService));
    doReturn(false).when(cps).lockSharedConfiguration();
    cps.createConfigurationResponse(null);
......

Reply:
you are right, that simplifies a lot the test case. Thanks!